runtime/caml/address_class.h

```diff
@@ -74,8 +74,12 @@
 
 #define Is_in_heap_or_young(a) (Classify_addr(a) & (In_heap | In_young))
 
+#ifdef NO_NAKED_POINTERS
+#define Is_in_value_area(a) 1
+#else
 #define Is_in_value_area(a)                                     \
   (Classify_addr(a) & (In_heap | In_young | In_static_data))
+#endif
```

Comment:
This change is, I think, less harmless than it seems. As is more clear in the following commits, it actually changes the behavior of hashing and comparison when it comes to out-of-heap values (such as black-colored blocks allocated by C libraries outside of the heap).

Even though I think this change in behavior is fine, it should be well advertised in (at least) a `Changes` entry.

Reply:
It is correct that hashing and comparison now work differently over naked pointers in classic mode and the corresponding out-of-heap pointers-with-a-black-header in no-naked-pointers heap.  (That was already the case before this PR.)  OK for documenting it in a Changes entry or elsewhere, but quite frankly the previous behavior wasn't terribly useful to begin with.